src/QueryBuilder/QueryViewer/QueryViewer.js

```diff
@@ -0,0 +1,124 @@
+import React, { memo, useEffect, useMemo } from 'react';
+import PropTypes from 'prop-types';
+import { Col, Row, Accordion, AccordionSet, MultiColumnList, Headline } from '@folio/stripes/components';
+import { PrevNextPagination, usePagination } from '@folio/stripes-acq-components';
+import QueryLoader from '../QueryLoader/QueryLoader';
+import { getTableMetadata } from '../helpers';
+
+const QueryViewer = ({
+  showPagination = true,
+  contentData = [],
+  limit = 100,
+  offset = 0,
+  onPageChange,
+  entityType,
+  headline,
+  headlineEnd,
+  visibleColumns,
+  onSetDefaultVisibleColumns,
+  onSetDefaultColumns,
+  accordionHeadline,
+  isInProgress,
+  inProgressTitle,
+  height,
+}) => {
+  const { changePage, pagination } = usePagination({ limit, offset });
+  const {
+    columnMapping,
+    defaultColumns,
+    defaultVisibleColumns,
+  } = useMemo(() => getTableMetadata(entityType), [entityType]);
+
+  useEffect(() => {
+    onPageChange?.(pagination);
```

Comment:
what is the use case for this?

Reply:
described here https://github.com/folio-org/ui-plugin-query-builder/pull/2#discussion_r1101518028